test/integration/integration_test.go

```diff
@@ -169,10 +170,10 @@ func FeatureContext(s *godog.Suite) {
 		getRoutingUrlAndVerifyHTTPResponse)
 
 	// steps for verifying config file content
-	s.Step(`JSON config file "([^"]*)" (contains|does not contain) key "(.*)" with value "(.*)"$`,
-		configContains)
-	s.Step(`JSON config file "([^"]*)" (contains|does not contain) key "(.*)"(.*)$`,
-		configContains)
+	s.Step(`^JSON config file "([^"]*)" (contains|does not contain) key "(.*)" with value "(.*)"$`,
```

Comment:
"s.Step(`^JSON config file "([^"]*)" (contains|does not contain) key "(.*)" with value matching "(.*)"$`"

might be more explicit and making clear in the feature file that we do regular expression matching.